packages/backend/src/tokens/request.ts

```diff
@@ -552,7 +527,16 @@ ${error.getFullMessage()}`,
       if (error?.cause?.reason) {
         refreshError = error.cause.reason;
       } else {
-        refreshError = RefreshTokenErrorReason.UnexpectedRefreshError;
+        refreshError = RefreshTokenErrorReason.UnexpectedSDKError;
+      }
+    } else {
+      if (request.method !== 'GET') {
+        refreshError = RefreshTokenErrorReason.NonEligibleNonGet;
+      } else if (!authenticateContext.refreshTokenInCookie) {
+        refreshError = RefreshTokenErrorReason.NonEligibleNoCookie;
+      } else {
+        //refresh error is not applicable if token verification error is not 'session-token-expired'
+        refreshError = null;
```

Comment:
🔧 For correctness and to make this bullet-proof, I'd suggest to actually put something in here. Do we ever expect this case to be reached? If so, why?

How about something like `session-token-expired-refresh-non-eligible-other`.

Perhaps now it's impossible for this path to be reached, but if the code changes in the future, this might not be true. Having a default value guards against such cases.

Reply:
Any value here would be ignored later. This is the case where we don't have a 'session token expired' case, so refresh flow is not used, that's why I used `null` here. Initially, I had no `else` statement here, because we don't really need the `refreshError` value.
This will end up being a handshake with `session-token-nbf` or `session-token-iat-in-the-future`, or else a signed out state with a relevant reason.